aea/test_tools/click_testing.py

```diff
@@ -56,7 +64,20 @@ def invoke(  # type: ignore
         exception: Optional[BaseException] = None
         exit_code = 0
 
-        with self.isolation(input=input, env=env, color=color) as outstreams:
+        if self.capfd:
+            message = """
+            Cannot use capfd in conjunction with `input`, `env` or `color`.
+            If you wish to enable their usage with capfd, you may patch the
+            CliRunner.isolation context manager. Three lines of code to set
+            back the newly assigned sys.stdin, sys.stdout and sys.stderr
+            immediately after they get initially re-assigned likely suffices
+            """
+            assert not (input or env or color), message  # nosec
```

Comment:
Never use `assert` statement outside of the tests

Reply:
this is a test tool, used in tests, which I why I opted for this. But I see your point, it is part of our framework and hence it is better to raise.